storage/store/chunks_queue.go

```diff
@@ -0,0 +1,175 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/chunks"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/operation"
+)
+
+// ChunksQueue stores a queue of chunk locators that assigned to me to verify.
+// Job consumers can read the locators as job from the queue by index.
+// Chunk locators stored in this queue are unique.
+type ChunksQueue struct {
+	db                storage.DB
+	chunkLocatorCache *Cache[uint64, *chunks.Locator] // cache for chunk locators, indexed by job index
+	storing           *sync.Mutex
```

Comment:
its not quite clear what this is locking. It doesn't look like its locking the cache nor the db because `AtIndex` just accesses both without a lock.

Reply:
The storing lock is to prevent dirty read of `ExistChunkLocator`. The Store method depends on the return value of this read. Without the lock, the returned value could be wrong if it's updated by another Store call on a different thread.